extensions/test/kokkos/types.cpp

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: 2017-2023 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <cstring>
+#include <Kokkos_Core.hpp>
+#include <sstream>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/mtx_io.hpp>
+#include <ginkgo/extensions/kokkos/spaces.hpp>
+#include <ginkgo/extensions/kokkos/types.hpp>
+
+
+#include "core/test/utils.hpp"
+
+
+using DefaultMemorySpace = Kokkos::DefaultExecutionSpace::memory_space;
+
+
+template <typename ValueType>
+class ArrayMapper : public ::testing::Test {
+protected:
+    using value_type = ValueType;
+
+    std::shared_ptr<gko::Executor> exec =
+        gko::ext::kokkos::create_default_executor();
+    gko::array<value_type> array = {exec, I<value_type>{1, 2, 3, 4}};
+};
+
+TYPED_TEST_SUITE(ArrayMapper, gko::test::ValueTypes, TypenameNameGenerator);
```

Comment:
also check whether changing ginkgo array affect kokkos value

Reply:
I'm now checking that both point to the same memory. Do you think that's enough?

Reply:
I think it should be enough for array case. for dense 2d with stride, checking the change is still neccessary for ensuring our mapping correctly (also represented row-major not column major in kokkos side)

Reply:
I'm also comparing the stride, sizes, and number of stored values, so that might be enough. Comparing the values directly, is a bit tricky, since they are in the device. So far I've failed to copy the Kokkos view to the host, but I will spend a bit more time trying that.